Packages/MIES/MIES_Utilities_File.ipf

```diff
@@ -222,96 +222,175 @@ Function/S GetUniqueSymbolicPath([string prefix])
-		for(fileName : allFilesInDir)
+		cdf = folders[i]
+		NewPath/Q/O/Z $workPath, cdf
 
-			GetFileFolderInfo/P=$pathName/Q/Z fileName
+		subFoldersList = IndexedDir($workPath, -1, 1, FILE_LIST_SEP); err = GetRTError(1)
+		WAVE/T subFolders = ListToTextWave(subFoldersList, FILE_LIST_SEP)
+		// add trailing colon
+		Multithread subFolders[] += ":"
 
-			if(V_Flag != 0)
-				// error querying file
-				continue
-			endif
+		ConcatenateWavesWithNoteIndex(folders, subFolders)
 
-			if(V_IsAliasShortcut)
-				fileOrPath = ResolveAlias(fileName, pathName = pathName)
+		subFilesList = IndexedFile($workPath, -1, "????", "????", FILE_LIST_SEP); err = GetRTError(1)
+		WAVE/T subFiles = ListToTextWave(subFilesList, FILE_LIST_SEP)
+		// make them absolute
+		Multithread subFiles[] = cdf + subFiles[p]
 
-				if(IsEmpty(fileOrPath))
-					// dead alias
-					continue
-				endif
+		if(resolveAliases)
+			[WAVE/T filesResolved, WAVE/T foldersResolved] = GetAllFilesAliasHelper(subFiles)
+		else
+			WAVE/T filesResolved = subFiles
+			WAVE/ZZ/T foldersResolved
+		endif
 
-				// redo the check
-				GetFileFolderInfo/P=$pathName/Q/Z fileOrPath
+		numFiles   = ConcatenateWavesWithNoteIndex(resultFiles, filesResolved)
+		numFolders = ConcatenateWavesWithNoteIndex(folders, foldersResolved)
+	endfor
 
-				if(V_Flag != 0)
-					// error querying file/folder pointed to by alias
-					continue
-				endif
-			endif
+	if(!numFiles)
+		return $""
+	endif
 
-			if(V_isFile)
-				allFiles = AddListItem(S_path, allFiles, FILE_LIST_SEP, Inf)
-			elseif(V_isFolder)
-				foldersFromAlias = AddListItem(S_path, foldersFromAlias, FILE_LIST_SEP, Inf)
-			else
-				ASSERT(0, "Unexpected file type")
-			endif
-		endfor
+	Redimension/N=(numFiles) resultFiles
+	Note/K resultFiles
+
+	if(needsFiltering)
+		WAVE/Z/T resultFilesFiltered = GrepTextWave(resultFiles, regex)
+	else
+		WAVE/T resultFilesFiltered = resultFiles
 	endif
 
-	AssertOnAndClearRTError()
-	folders = IndexedDir($pathName, -1, 1, FILE_LIST_SEP); err = GetRTError(1)
-	folders = folders + foldersFromAlias
-	WAVE/T wFolders = ListToTextWave(folders, FILE_LIST_SEP)
-	for(folder : wFolders)
+	return resultFilesFiltered
+End
 
-		subFolderPathName = GetUniqueSymbolicPath()
+static Function [WAVE/Z/T filesResolved, WAVE/Z/T foldersResolved] GetAllFilesAliasHelper(WAVE/T files)
 
-		NewPath/Q/O $subFolderPathName, folder
-		files = GetAllFilesRecursivelyFromPath(subFolderPathName, extension = extension)
-		KillPath/Z $subFolderPathName
+	string fileName, fileOrPath
+	variable numFiles, numFolders
 
-		if(!isEmpty(files))
-			allFiles = AddListItem(RemoveEnding(files, FILE_LIST_SEP), allFiles, FILE_LIST_SEP, Inf)
+#ifdef WINDOWS
+	WAVE/Z/T aliasFiles = GrepTextWave(files, "\.lnk$")
+
+	if(!WaveExists(aliasFiles))
+		return [files, $""]
+	endif
+#else
+	WAVE aliasFiles = files
+#endif // WINDOWS
+
+	Make/T/FREE/N=(MINIMUM_WAVE_SIZE) filesResolved, foldersResolved
+
+	for(fileName : aliasFiles)
+
+		GetFileFolderInfo/Q/Z fileName
```

Comment:
What is a bit off here from the logic:

The function gets files as input.

If GetFileFolderInfo here resolves and V_IsAliasShortcut == 0 then it can directly appended to filesResolved.

The V_isFile / V_isFolder handling only needs to be in the branch when V_IsAliasShortcut is true.

I would also remark here that this GetFileFolderInfo call is only to get V_IsAliasShortcut